src/read.jl

```diff
@@ -446,12 +440,12 @@ function internal_xml_file_read(xf::XLSXFile, filename::String) :: EzXML.Documen
     if !internal_xml_file_isread(xf, filename)
         @assert isopen(xf) "Can't read from a closed XLSXFile."
         file_not_found = true
-        for f in xf.io.files
-            if f.name == filename
+        for f in ZipArchives.zip_names(xf.io)
+            if f == filename
                 xf.files[filename] = true # set file as read
 
                 try
-                    xf.data[filename] = EzXML.readxml(f)
+                    xf.data[filename] = EzXML.parsexml(ZipArchives.zip_readentry(xf.io, f, String))
                 catch err
                     @error("Failed to parse internal XML file `$filename`")
                     rethrow()
```

Comment:
The whole for loop and `file_not_found` check can be simplified to:

```julia
try
    xf.data[filename] = EzXML.parsexml(ZipArchives.zip_readentry(xf.io, filename))
catch err
    @error("Failed to parse internal XML file `$filename`")
    rethrow()
end
```

Because `zip_readentry` will take care of looping through the entry names, reading a matching entry, or erroring if the filename isn't found.

Reply:
Thank you very much forthese suggestions. Where possible, I've accepted them here, as you can see.

Your last suggestion also seems to need the line
```
xf.files[filename] = true # set file as read
```
in the `try` block, otherwise the file is recorded as not present.
I've added this to my fork and will figure out how to include it here.